tgui/packages/tgui/interfaces/LightController.tsx

```diff
@@ -0,0 +1,319 @@
+import { useBackend, useLocalState } from '../backend';
+import { round } from '../../common/math';
+import { classes } from '../../common/react';
+import { Box, Button, Knob, Section, Slider, Stack, Tabs } from '../components';
+import { Window } from '../layouts';
+
+const NORTH = 1;
+const SOUTH = 2;
+const EAST = 4;
+const WEST = 8;
+
+type LightDetails = {
+  name: string;
+  color: string;
+  power: number;
+  range: number;
+  angle: number;
+};
+
+type TemplateID = string;
+
+type LightTemplate = {
+  light_info: LightDetails;
+  description: string;
+  id: TemplateID;
+};
+
+
+interface CategoryList {
+  [key: string]: TemplateID[];
+}
+
+type Data = {
+  on: boolean;
```

Comment:
```suggestion
  on: BooleanLike;
```

you'll need to import this as well